Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
```

Comment:
Yeh, I'd rather rename it to e.g. `CoreFormuleTap` or something when it's actually a tap. My initial thought from the name was that `homebrew/versions` was a `CoreTap`.